kubectl-fdb/cmd/buggify_crash_loop_test.go

```diff
@@ -161,5 +163,231 @@ var _ = Describe("[plugin] buggify crash-loop instances command", func() {
+						}),
+					Entry("Adding multiple instances.",
+						testCase{
+							ContainerName:                testContainerName,
+							Instances:                    []string{"test-storage-1", "test-storage-2"},
+							ExpectedInstancesInCrashLoop: []fdbv1beta2.ProcessGroupID{"storage-1", "storage-2"},
+						}),
+				)
+
+			})
+
+			When("crash-loop container list contains the input container", func() {
+				BeforeEach(func() {
+					crashLoopContainerObj := fdbv1beta2.CrashLoopContainerObject{
+						ContainerName: testContainerName,
+						Targets:       []fdbv1beta2.ProcessGroupID{"storage-1"},
+					}
+					cluster.Spec.Buggify.CrashLoopContainers = append(cluster.Spec.Buggify.CrashLoopContainers, crashLoopContainerObj)
+				})
+
+				DescribeTable("should add all targeted processes to crash-loop container list",
+					func(tc testCase) {
+						Expect(len(cluster.Spec.Buggify.CrashLoopContainers)).To(BeNumerically("==", 1))
+						Expect(updateCrashLoopContainerList(k8sClient, clusterName, tc.ContainerName, tc.Instances, namespace, false, false, false)).NotTo(HaveOccurred())
+
+						var resCluster fdbv1beta2.FoundationDBCluster
+						Expect(k8sClient.Get(context.Background(), client.ObjectKey{
+							Namespace: namespace,
+							Name:      clusterName,
+						}, &resCluster)).NotTo(HaveOccurred())
+						Expect(len(resCluster.Spec.Buggify.CrashLoopContainers)).To(BeNumerically("==", 1))
+						for _, crashLoopContainerObj := range resCluster.Spec.Buggify.CrashLoopContainers {
+							if crashLoopContainerObj.ContainerName != tc.ContainerName {
+								continue
+							}
+							Expect(tc.ExpectedInstancesInCrashLoop).To(ContainElements(crashLoopContainerObj.Targets))
+							Expect(len(tc.ExpectedInstancesInCrashLoop)).To(BeNumerically("==", len(crashLoopContainerObj.Targets)))
+						}
+					},
+					Entry("Adding same instance.",
+						testCase{
+							ContainerName:                testContainerName,
+							Instances:                    []string{"test-storage-1"},
+							ExpectedInstancesInCrashLoop: []fdbv1beta2.ProcessGroupID{"storage-1"},
+						}),
+					Entry("Adding single different instance.",
+						testCase{
+							ContainerName:                testContainerName,
+							Instances:                    []string{"test-storage-2"},
+							ExpectedInstancesInCrashLoop: []fdbv1beta2.ProcessGroupID{"storage-1", "storage-2"},
+						}),
+					Entry("Adding multiple instances.",
+						testCase{
+							ContainerName:                testContainerName,
+							Instances:                    []string{"test-storage-1", "test-storage-2", "test-storage-3"},
+							ExpectedInstancesInCrashLoop: []fdbv1beta2.ProcessGroupID{"storage-1", "storage-2", "storage-3"},
+						}),
+				)
+			})
+
+			When("crash-loop container list contains doesn't the input container", func() {
+				BeforeEach(func() {
+					crashLoopContainerObj := fdbv1beta2.CrashLoopContainerObject{
+						ContainerName: "test-container-2",
```

Comment:
Can we use the sidecar or main container name here instead of using a custom value?